packages/in_app_purchase/in_app_purchase_android/example/lib/main.dart

```diff
@@ -22,7 +22,9 @@ void main() {
   runApp(_MyApp());
 }
 
-const bool _kAutoConsume = true;
+// Auto-consume must be true on iOS.
+// To try without auto-consume on another platform, change `true` to `false` here.
+final bool _kAutoConsume = Platform.isIOS || true;
```

Comment:
And the `Platform.iOS` option.

Reply:
If we remove `Platform.isIOS` then we are back to the **redundant_argument_values** linter conflict.

So I removed the _kAutoConsume redundant argument from call side, committed this change, but now i realised that the _kAutoConsume is used in another place:

```
if (!_kAutoConsume && purchaseDetails.productID == _kConsumableId) {
    final InAppPurchaseAndroidPlatformAddition addition = InAppPurchasePlatformAddition.instance! as InAppPurchaseAndroidPlatformAddition;
    await addition.consumePurchase(purchaseDetails);
}
```

So leaving Platform.isIOS is some kind of workaround for having the ability to turn on/off the auto consume mode from one place and not braking the lint rule.

Reply:
In the Android case we have a specific reason we are passing that argument (to allow for the variable changing it in one place), so we should just `// ignore` the warning at the call site.